sources/ClangSharp/Interop.Extensions/clang.ResolveLibrary.cs

```diff
@@ -44,6 +44,11 @@ private static bool TryResolveClang(Assembly assembly, DllImportSearchPath? sear
                 return true;
             }
 
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) && NativeLibrary.TryLoad("clang-10", assembly, searchPath, out nativeLibrary))
```

Comment:
`clang-10` is a different binary, could you clarify why it is needed?

Reply:
I am using the upstream libclang 10 library located at https://apt.llvm.org/ for ubuntu 18.04. The libclang 10 library is located at `/usr/lib/llvm-10/lib/libclang-10.so.1`, but for Linux it is looking for libclang.so.10.

I get the below error if I don't have this patch in the codebase.
```
Unhandled exception: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.DllNotFoundException: Unable to load shared library 'libclang' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: liblibclang: cannot open shared object file: No such file or directory
   at ClangSharp.Interop.clang.createIndex(Int32 excludeDeclarationsFromPCH, Int32 displayDiagnostics)
   at ClangSharp.Interop.CXIndex.Create(Boolean excludeDeclarationsFromPch, Boolean displayDiagnostics) in /home/jerry/dotnet/AspCoreNginxInProcess/modules/ClangSharp/sources/ClangSharp/Interop.Extensions/CXIndex.cs:line 30
   at ClangSharp.PInvokeGenerator..ctor(PInvokeGeneratorConfiguration config, Func`2 outputStreamFactory) in /home/jerry/dotnet/AspCoreNginxInProcess/modules/ClangSharp/sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.cs:line 40
   at ClangSharp.Program.Run(InvocationContext context) in /home/jerry/dotnet/AspCoreNginxInProcess/modules/ClangSharp/sources/ClangSharpPInvokeGenerator/Program.cs:line 283
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at System.CommandLine.Invocation.ModelBindingCommandHandler.InvokeAsync(InvocationContext context)
   at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseParseErrorReporting>b__20_0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass15_0.<<UseHelp>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass24_0.<<UseVersionOption>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass22_0.<<UseTypoCorrections>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseSuggestDirective>b__21_0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseParseDirective>b__19_0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseDebugDirective>b__11_0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<RegisterWithDotnetSuggest>b__10_0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass13_0.<<UseExceptionHandler>b__0>d.MoveNext()
```

Reply:
The file being looked for should be `libclang-10.so` or `libclang-10` then. Looking for `clang-10` risks it matching the clang executable instead, which would be undesirable.

Reply:
I will fix the patch and upload to this pull request.